sdk/identity/Azure.Identity/src/CredentialDiagnosticScope.cs

```diff
@@ -14,51 +14,69 @@ namespace Azure.Identity
         private readonly string _name;
         private readonly DiagnosticScope _scope;
         private readonly TokenRequestContext _context;
+        private readonly IScopeHandler _scopeHandler;
 
-        public CredentialDiagnosticScope(string name, DiagnosticScope scope, TokenRequestContext context)
+        public CredentialDiagnosticScope(string name, TokenRequestContext context, IScopeHandler scopeHandler)
         {
             _name = name;
-
-            _scope = scope;
-
+            _scope = scopeHandler.CreateScope(name);
             _context = context;
+            _scopeHandler = scopeHandler;
         }
 
         public void Start()
         {
-            _scope.Start();
+            AzureIdentityEventSource.Singleton.GetToken(_name, _context);
+            _scopeHandler.Start(_name, _scope);
         }
 
         public AccessToken Succeeded(AccessToken token)
         {
             AzureIdentityEventSource.Singleton.GetTokenSucceeded(_name, _context, token.ExpiresOn);
-
             return token;
         }
 
         public Exception FailWrapAndThrow(Exception ex)
         {
-            if (ex is OperationCanceledException || ex is AuthenticationFailedException)
+            var wrapped = TryWrapException(ref ex);
+            RegisterFailed(ex);
+
+            if (!wrapped)
             {
-                var info = ExceptionDispatchInfo.Capture(ex);
-                RegisterFailed(ex);
-                info.Throw();
+                ExceptionDispatchInfo.Capture(ex).Throw();
             }
 
-            ex = new AuthenticationFailedException($"{_name.Substring(0, _name.IndexOf('.'))} authentication failed.", ex);
-            RegisterFailed(ex);
             throw ex;
         }
 
         private void RegisterFailed(Exception ex)
         {
             AzureIdentityEventSource.Singleton.GetTokenFailed(_name, _context, ex);
-            _scope.Failed(ex);
+            _scopeHandler.Fail(_name, _scope, ex);
         }
 
-        public void Dispose()
+        private bool TryWrapException(ref Exception exception)
         {
-            _scope.Dispose();
+            if (exception is OperationCanceledException || exception is AuthenticationFailedException)
+            {
+                return false;
+            }
+
+            if (exception is AggregateException aex)
```

Comment:
Additional handling for `AggregateException` is required for methods like `VisualStudioCredential.RunProcessesAsync`

Reply:
Why do we need to handle this here vs. directly in `VisualStudioCredential.RunProcessAsync`. I'm worried that this is trying to handle too many cases.